zerver/lib/retention.py

```diff
@@ -272,14 +274,17 @@ def archive_personal_and_huddle_messages(realm: Realm, chunk_size: int=MESSAGE_B
     logger.info("Done. Archived %s messages", message_count)
 
 def archive_stream_messages(realm: Realm, chunk_size: int=MESSAGE_BATCH_SIZE) -> None:
-    logger.info("Archiving stream messages for realm " + realm.string_id)
     # We don't archive, if the stream has message_retention_days set to -1,
     # or if neither the stream nor the realm have a retention policy.
     streams = Stream.objects.select_related("recipient").filter(
         realm_id=realm.id).exclude(message_retention_days=-1)
     if not realm.message_retention_days:
         streams = streams.exclude(message_retention_days__isnull=True)
 
+    if not streams.exists():
```

Comment:
Doesn't this generate an extra round trip when we actually want to use the data?  We could instead rename `streams` to `query` above, and then say `streams = list(query)` and then not `if not streams: return`.

Reply:
Oh yeah, that's definitely a good idea, fixed!